### ISSUE TITLE
[Bug] Flash attention prefill always passes attention_mask=None, causing quality drop when batch_size > 1

### ISSUE THREAD

**Issue by QuiverDance** [jy-yuan/KIVI#46]
When `cfg._flash_attn_2_enabled = True`, the flash attention path in `MistralFlashAttention_KIVI` uses `_flash_attention_forward` in the prefill branch like this:

'''
query_states = query_states.to(target_dtype)
key_states_repeat = key_states_repeat.to(target_dtype)
value_states_repeat = value_states_repeat.to(target_dtype)

attn_output = self._flash_attention_forward(
    query_states.transpose(1, 2),
    key_states_repeat.transpose(1, 2),
    value_states_repeat.transpose(1, 2),
    None,          <- attention_mask is always None here
    q_len,
    dropout=0.0,
)
'''

The problem is that attention_mask is always forced to None in the prefill path.

When batch_size > 1, inputs are padded to the same length. Because the mask is not passed, those padding tokens are included in the internal softmax computation in flash attention. As a result, the attention scores are “contaminated” by padding positions instead of being computed only over real (non-pad) tokens.

In practice,
For batch_size = 1: MMLU and other benchmarks give some baseline score.
For batch_size > 1: the scores drop significantly with flash attention enabled.

If I modify this call to pass attention_mask into _flash_attention_forward (instead of None), the benchmark scores for batch_size = 1 and batch_size > 1 become consistent again.

Thus, _flash_attention_forward should receive the actual attention_mask, but it is currently hard-coded to None, which breaks correctness when there is padding in the batch.
